internal/app/machined/internal/runtime/initializer/baremetal/baremetal.go

```diff
@@ -0,0 +1,70 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package baremetal
+
+import (
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/talos-systems/talos/internal/app/machined/internal/platform"
+	"github.com/talos-systems/talos/internal/pkg/installer"
+	"github.com/talos-systems/talos/internal/pkg/kernel"
+	"github.com/talos-systems/talos/internal/pkg/mount"
+	"github.com/talos-systems/talos/internal/pkg/mount/manager"
+	"github.com/talos-systems/talos/internal/pkg/mount/manager/owned"
+	"github.com/talos-systems/talos/pkg/constants"
+	"github.com/talos-systems/talos/pkg/userdata"
+)
+
+// Baremetal represents an initializer that performs a full installation to a
+// disk.
+type Baremetal struct{}
+
+// Initialize implements the Initializer interface.
+func (b *Baremetal) Initialize(platform platform.Platform, data *userdata.UserData) (err error) {
+	var endpoint *string
+	if endpoint = kernel.ProcCmdline().Get(constants.KernelParamUserData).First(); endpoint == nil {
+		return errors.Errorf("failed to find %s in kernel parameters", constants.KernelParamUserData)
+	}
+
+	cmdline := kernel.NewDefaultCmdline()
+	cmdline.Append("initrd", filepath.Join("/", "default", "initramfs.xz"))
+	cmdline.Append(constants.KernelParamPlatform, strings.ToLower(platform.Name()))
```

Comment:
`strings.ToLower(platform.Name())` forced the rename of the `bare-metal` platform. This was a good time to align some of the other outstanding TODOs (like rename `googlecloud`  to `gcp`. GCP is technically the correct name if we are going to name the platform after the over cloud provider service (e.g AWS instead of EC2, Azure instead of Virtual Machines).